src/ntfs/write.rs

```diff
@@ -179,6 +177,229 @@ pub fn write_fares_v1(
     Ok(())
 }
 
+fn write_fares_v1_from_v2(
+    base_path: &path::Path,
+    tickets: &CollectionWithId<Ticket>,
+    ticket_uses: &CollectionWithId<TicketUse>,
+    ticket_prices: &Collection<TicketPrice>,
+    ticket_use_perimeters: &Collection<TicketUsePerimeter>,
+    ticket_use_restrictions: &Collection<TicketUseRestriction>,
+) -> Result<()> {
+    fn has_no_transfer(ticket_uses: &CollectionWithId<TicketUse>, ticket_use_id: &String) -> bool {
+        if let Some(ticket_use) = ticket_uses.get(ticket_use_id) {
+            if (ticket_use.max_transfers.is_none() || ticket_use.max_transfers == Some(0))
+                && ticket_use.boarding_time_limit.is_none()
+                && ticket_use.alighting_time_limit.is_none()
+            {
+                return true;
+            }
+        }
+        false
+    }
+
+    let mut prices_v1: Vec<PriceV1> = vec![];
+    let mut fares_v1: Vec<FareV1> = vec![];
+
+    // Conversion of OD fares on specific lines
+
+    for ticket_use_restriction in ticket_use_restrictions
```

Comment:
I think I would extract this for-loop into a function. So you'd have a `convert_od_on_line()` function, then another function `convert_flat_fare_on_network()` and you could even put the section of the specification for each in a RustDoc comment.